sotodlib/toast/ops/splits.py

```diff
@@ -544,34 +592,57 @@ def _exec(self, data, detectors=None, **kwargs):
 
         # Loop over splits
         for split_name, spl in self._split_obj.items():
-            log.info_rank(f"Running Split '{split_name}'", comm=data.comm.comm_world)
             # Set mapmaker name based on split and the name of this
             # Splits operator.
             mname = f"{self.name}_{split_name}"
             self.mapmaker.name = mname
+            log.info_rank(
+                f"Running Split '{split_name}'", comm=data.comm.comm_world
+            )
 
             # Apply this split
             for ob in data.obs:
                 spl.create_split(ob)
 
-            # Set mapmaking tools to use the current split interval list
-            map_binner.pixel_pointing.view = spl.split_intervals
-            if not map_binner.full_pointing:
-                # We are not using full pointing and so we clear the
-                # residual pointing for this split
-                toast.ops.Delete(
-                    detdata=[
-                        map_binner.pixel_pointing.pixels,
-                        map_binner.stokes_weights.weights,
-                        map_binner.pixel_pointing.detector_pointing.quats,
-                    ],
-                ).apply(data)
-
-            # Run mapmaking
-            self.mapmaker.apply(data)
-
-            # Write
-            self.write_splits(data, split_name=split_name)
+            if self.skip_existing and self.splits_exist(data, split_name=split_name):
+                log.info_rank(
+                    f"All outputs for split '{split_name}' exist, skipping...",
+                    comm=data.comm.comm_world,
+                )
+            else:
+                if self.splits_as_flags:
+                    # Split is applied through flagging, not as a view
+                    # Flag samples outside the intervals by prefixing '~'
+                    # to the view name
+                    FlagIntervals(
+                        shared_flags=self.shared_flags,
+                        shared_flag_bytes=1,
+                        view_mask=[
+                            (f"~{spl.split_intervals}", np.uint8(self.shared_flag_mask))
+                        ],
+                        reset=True,
+                    ).apply(data)
+                    map_binner.shared_flag_mask |= self.shared_flag_mask
```

Comment:
Inside the `Split` base class, when a split is created a interval list is created and the per-detector flags are saved.  When the split is removed, that interval list is purged and the per-detector flags are restored.

If you are applying the `FlagIntervals` operator, that will modify the shared flags and leave it in a different state at the end.  I think in order to do this cleanly, you would need to save the shared_flag values, overwrite them here for each split, and then restore them when each split is removed.

Reply:
Good point. Just modifying a user-specified flag bit is not enough. I added saving and restoring of the shared flags. I also expanded on the existing unit test to confirm that intervals and flags produce the same output maps.